syntax = "proto3";

service Peer {
    // Allows a new broker to request head to join cluster
    rpc HandleJoinCluster(JoinClusterRequest) returns (JoinClusterResponse);
    // Notify all other brokers of a new broker
    rpc HandleReportNewBroker(ReportNewBrokerRequest) returns (ReportNewBrokerResponse);
}

// Request containing address and port of sender (new broker)
message JoinClusterRequest {
    string address = 1;
    string port = 2;
}

// Empty response for now
message JoinClusterResponse {}

// Request containing address and port of new broker
message ReportNewBrokerRequest {
    string address = 1;
    string port = 2;
}

// Empty response for now
message ReportNewBrokerResponse {}
